pkgs/development/libraries/qt-5/modules/qtbase.nix

```diff
@@ -135,10 +135,15 @@ stdenv.mkDerivation {
           -e 's#sdk_val=$(/usr/bin/xcrun -sdk $sdk -find $(echo $val | cut -d \x27 \x27 -f 1))##' \
           -e 's#val=$(echo $sdk_val $(echo $val | cut -s -d \x27 \x27 -f 2-))##' \
           ./configure
-          substituteInPlace ./mkspecs/common/mac.conf \
-              --replace "/System/Library/Frameworks/OpenGL.framework/" "${darwin.apple_sdk.frameworks.OpenGL}/Library/Frameworks/OpenGL.framework/"
-          substituteInPlace ./mkspecs/common/mac.conf \
-              --replace "/System/Library/Frameworks/AGL.framework/" "${darwin.apple_sdk.frameworks.AGL}/Library/Frameworks/AGL.framework/"
+      substituteInPlace ./mkspecs/common/mac.conf \
+        --replace "/System/Library/Frameworks/OpenGL.framework/" "${darwin.apple_sdk.frameworks.OpenGL}/Library/Frameworks/OpenGL.framework/"
+      substituteInPlace ./mkspecs/common/mac.conf \
+        --replace "/System/Library/Frameworks/AGL.framework/" "${darwin.apple_sdk.frameworks.AGL}/Library/Frameworks/AGL.framework/"
+
+      # Recent Qt releases use SDK (and Objective-C) features beyond
+      # the 10.10 available in nixpkgs
+      sed 's/NSEnumerator<NSWindow\*>/NSEnumerator/' -i ./src/plugins/platforms/cocoa/qcocoawindow.mm
```

Comment:
I prefer using substituteInPlace for literals so you don't have to worry about escaping stuff.